aten/src/ATen/native/cuda/Blas.cpp

```diff
@@ -167,6 +167,21 @@ Tensor& addmm_out_cuda_impl(Tensor& result, const Tensor& self, const Tensor& ma
   at::ScalarType scalar_type = self.scalar_type();
   c10::MaybeOwned<Tensor> self_;
   if (&result != &self) {
+#if defined(USE_ROCM) && ROCM_VERSION >= 50600
+    if (std::getenv("USE_HIPBLASLT")){
+      hipDeviceProp_t* prop = at::cuda::getDeviceProperties(self.device().index());
+      std::string device_arch = prop->gcnArchName;
+      static const std::string hipblaslt_supported_arch = "gfx90a";
+      size_t substring = device_arch.find(hipblaslt_supported_arch);
+      TORCH_CHECK(substring != std::string::npos, "Attempting to use HIPBlasLT on a unsupported architecture!");
```

Comment:
Another approach is
1. Do not check `USE_HIPBLASLT` at the beginning
2. Default to hipblasLt on MI200 unless `USE_HIPBLASLT=0` is set.

After all this depends on the client's request, or we can simply match CUDA's behavior.

Reply:
So the situation is we want the user to opt-in to using hipblaslt because we don't want perf regressions. Our hipblaslt implementation is still under optimization phase, hence why we decided to not have it set as default.